src/main/java/com/rcv/ElectionConfig.java

```diff
@@ -59,23 +62,119 @@ boolean validate() {
+        } else {
+          cvrFilePathSet.add(source.getFilePath());
+        }
+
+        if (source.getFirstVoteColumnIndex() == null) {
+          isValid = false;
+          Logger.executionLog(
+              Level.SEVERE, "firstVoteColumnIndex is required: %s", source.getFilePath());
+        } else if (source.getFirstVoteColumnIndex() < 0
+            || source.getFirstVoteColumnIndex() > 1000) {
+          isValid = false;
+          Logger.executionLog(
+              Level.SEVERE,
+              "firstVoteColumnIndex must be from 0 to 1000: %s",
+              source.getFilePath());
+        }
+
+        if (source.getIdColumnIndex() != null
+            && (source.getIdColumnIndex() < 0 || source.getIdColumnIndex() > 1000)) {
+          isValid = false;
+          Logger.executionLog(
+              Level.SEVERE, "idColumnIndex must be from 0 to 1000: %s", source.getFilePath());
+        }
+
+        if (isTabulateByPrecinctEnabled()) {
+          if (source.getPrecinctColumnIndex() == null) {
+            isValid = false;
+            Logger.executionLog(
+                Level.SEVERE,
+                "precinctColumnIndex is required when tabulateByPrecinct is enabled: %s",
+                source.getFilePath());
+          } else if (source.getPrecinctColumnIndex() < 0
+              || source.getPrecinctColumnIndex() > 1000) {
+            isValid = false;
+            Logger.executionLog(
+                Level.SEVERE,
+                "precinctColumnIndex must be from 0 to 1000: %s",
+                source.getFilePath());
+          }
+        }
+      }
+    }
+
+    HashSet<String> candidateNameSet = new HashSet<>();
+    HashSet<String> candidateCodeSet = new HashSet<>();
+    for (Candidate candidate : rawConfig.candidates) {
+      if (candidate.getName() == null || candidate.getName().isEmpty()) {
+        isValid = false;
+        Logger.executionLog(Level.SEVERE, "Name is required for each candidate.");
+      } else if (candidateNameSet.contains(candidate.getName())) {
+        isValid = false;
+        Logger.executionLog(
+            Level.SEVERE, "Duplicate candidate names are not allowed: %s", candidate.getName());
+      } else {
+        candidateNameSet.add(candidate.getName());
+      }
+
+      if (candidate.getCode() != null && !candidate.getCode().isEmpty()) {
+        if (candidateCodeSet.contains(candidate.getCode())) {
+          isValid = false;
+          Logger.executionLog(
+              Level.SEVERE, "Duplicate candidate codes are not allowed: %s", candidate.getCode());
+        } else {
+          candidateCodeSet.add(candidate.getCode());
+        }
+      }
+    }
+
+    if (candidateCodeSet.size() > 0 && candidateCodeSet.size() != candidateNameSet.size()) {
+      isValid = false;
+      Logger.executionLog(
+          Level.SEVERE,
+          "If candidate codes are used, a unique code is required for each candidate.");
+    }
+
+    if (getNumDeclaredCandidates() < 1) {
```

Comment:
`== 0` seems more intuitive

Reply:
Maybe, but < 1 covers more bases, e.g. if in future changes some bug slips in that allows getNumDeclaredCandidates() to return a negative number.